### PR TITLE
[WIP] Feature: Images Upload

### PR DESCRIPTION
**Pull request by suezzo**

### CODE REVIEW

File: src/components/ProgressBar/ProgressBar.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.progress-bar {
+  position: fixed;
+  z-index: 2147483647;
```
Comment (fstodulski) [inline src/components/ProgressBar/ProgressBar.scss]: you dont need such big index...
use 1001 - my modal have 1000

File: src/components/ProgressBar/ProgressBar.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  top: 0;
+  left: 0px;
```
Comment (fstodulski) [inline src/components/ProgressBar/ProgressBar.scss]: no need to use 'px' if you have 0 value

File: src/services/asset.ts [hunk 3]
```diff
@@ -13,3 +13,21 @@ export function getAsset(userId: number, typeId: number): Promise<IAsset> {
 
   return request;
 }
+
+export function uploadAsset(params: any): Promise<IAsset> {
+  const { assetType, files } = params;
+  console.log(params);
```
Comment (fstodulski) [inline src/services/asset.ts]: remove console.log

File: src/services/asset.ts [hunk 4]
```diff
@@ -13,3 +13,21 @@ export function getAsset(userId: number, typeId: number): Promise<IAsset> {
 
   return request;
 }
+
+export function uploadAsset(params: any): Promise<IAsset> {
```
Comment (fstodulski) [inline src/services/asset.ts]: params should have type

File: src/services/asset.ts [hunk 5]
```diff
@@ -13,3 +13,21 @@ export function getAsset(userId: number, typeId: number): Promise<IAsset> {
...
+  const { assetType, files } = params;
+  console.log(params);
+  const formData = new FormData();
+  formData.append('file', files[0]);
+
+  const fetch = new Fetch();
```
Comment (fstodulski) [inline src/services/asset.ts]: Fetch is an abstract class - you dont need to create new instance of it.
just use Fetch.request(***)

File: src/stores/Assets.ts [hunk 6]
```diff
@@ -1,16 +1,29 @@
...
   @observable
   avatar: IAsset = {
-    filePath: ''
+    filePath: '/default-image-square.png'
   } as IAsset;
+
+  @action
+  set(type: string, url: string) {
+    const parsedUrl = `${host}${url}`;
+
```
Comment (fstodulski) [inline src/stores/Assets.ts:20]: there is no else statement:
Switch could handle it.

**Review by fstodulski — CHANGES_REQUESTED** [review]
Check comments

**Review by stanimirtt — APPROVED** [review]

**Review by fstodulski — APPROVED** [review]
👍

**Pull request merged**